lib/cpp/src/thrift/transport/TServerSocket.cpp

```diff
@@ -284,7 +284,7 @@ void TServerSocket::listen() {
   hints.ai_family = PF_UNSPEC;
   hints.ai_socktype = SOCK_STREAM;
   hints.ai_flags = AI_PASSIVE | AI_ADDRCONFIG;
-  sprintf(port, "%d", port_);
+  snprintf(port, sizeof("65535"), "%d", port_);
```

Comment:
I think the allocation of the buffer for port is just as bizarre, but I wanted to remain consistent.  Let's see if the build passes; if it fails and I need to rebase on master to kick another because of a transient issue, I can make the changes you suggest.